src/error.rs

```diff
@@ -0,0 +1,40 @@
+use actix_web;
+use actix_web::{HttpResponse, ResponseError};
+use actix_web::http::StatusCode;
+use derive_more::Display;
+
+#[derive(Display, Debug)]
+pub enum Error {
+    #[display(fmt = "Query validate error: {}", _0)]
+    Validate(validator::ValidationErrors),
+    #[display(fmt = "Query deserialize error: {}", _0)]
+    Deserialize(serde_urlencoded::de::Error),
```

Comment:
Can we union `Deserialize` and `DeserializeJson`?
I think it's should be a single error with inner Enum.

Reply:
Yep! Check it out pls